noir/noir-repo/compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -52,12 +52,18 @@ struct SharedContext {
     /// There can be Brillig functions specified in SSA which do not act as
     /// entry points in ACIR (e.g. only called by other Brillig functions)
     /// This mapping is necessary to use the correct function pointer for a Brillig call.
-    brillig_generated_func_pointers: BTreeMap<FunctionId, u32>,
+    /// This uses the brillig parameters in the map since using slices with different lengths
+    /// needs to create different brillig entrypoints
+    brillig_generated_func_pointers: BTreeMap<(FunctionId, Vec<BrilligParameter>), u32>,
```

Comment:
Feels a little cumbersome to match on all the parameters when we are only worried about slices. Granted I realize this was a quick fix and thanks for resolving the issue.